packages/core/src/codewhisperer/commands/basicCommands.ts

```diff
@@ -417,11 +417,11 @@ export const registerToolkitApiCallback = Commands.declare(
             getLogger().info(`Trying to register toolkit callback. Toolkit is installed.`)
             if (toolkitApi) {
                 // when this command is executed by AWS Toolkit activation
-                _toolkitApi = toolkitApi
+                _toolkitApi = toolkitApi.getApi(VSCODE_EXTENSION_ID.amazonq)
             } else {
                 // when this command is executed by Amazon Q activation
                 const toolkitExt = vscode.extensions.getExtension(VSCODE_EXTENSION_ID.awstoolkit)
-                _toolkitApi = toolkitExt?.exports
+                _toolkitApi = toolkitExt?.exports.getApi(VSCODE_EXTENSION_ID.amazonq)
             }
             if (_toolkitApi) {
                 await registerToolkitApiCallbackOnce()
```

Comment:
(off topic): not sure Toolkit should be calling its own public API (that would also lead to some weird & unnecessary telemetry)

Reply:
Here this piece of code is invoked by Amazon Q. Toolkit won't be making API calls against itself.